src/values/ObjectValue.js

```diff
@@ -570,12 +570,17 @@ export default class ObjectValue extends ConcreteValue {
       if (realm.react.enabled && realm.react.reactProps.has(this)) {
         realm.react.reactProps.add(template);
       }
-      let result = AbstractValue.createTemporalFromBuildFunction(this.$Realm, ObjectValue, [template], ([x]) => x, {
+      let temporalArgs = [template];
+      let result = AbstractValue.createTemporalFromBuildFunction(this.$Realm, ObjectValue, temporalArgs, ([x]) => x, {
         skipInvariant: true,
         isPure: true,
       });
       invariant(result instanceof AbstractObjectValue);
       result.values = new ValuesDomain(template);
+      // Store the args for the temporal so we can easily clone
+      // and reconstruct the temporal at another point, rather than
+      // mutate the existing temporal
+      realm.temporalAliasArgs.set(result, temporalArgs);
```

Comment:
Why use a map to get at the arguments that are already stored in result and that can be obtained with result.args?

Reply:
They're not on the `AbstractValue`, rather they are the temporal generator node `args`. If the abstract value and the generator node both had the same args, this entire thing would be vastly easier to do :)

Reply:
Another way would be to link the generator entry to the abstract value, in fact that might be a general easier approach. In the other PR https://github.com/facebook/prepack/pull/2148, which this is a small part of, we also need to know the additional args when the generator entry was created so we can clone and re-create the generator entries. So having a link from an abstract value, and it's original temporal entry would be highly beneficial. If not, and you feel this is a bad approach, how might we do this? @sebmarkbage @NTillmann

Reply:
OK, looking at the code of generator.deriveAbstract I can't off hand see a compelling reason why the arguments should not also be stored on the abstract value. Perhaps there is some subtle reason arising from the structure of the serializer, so @NTillmann would have to weigh in on this.

Reply:
@hermanventer Good point. I've created a PR to change to that approach https://github.com/facebook/prepack/pull/2199. I tried to adds the args (to see what would happen) and it appears to break the serializer output and we run into invariants, so I guess the derived abstract values don't have args on the value for structure reasons like you suggest.